### ISSUE TITLE
Fix false negatives for negative values in shorthand-property-no-redundant-values

### ISSUE THREAD

**Issue by ssivanatarajan** [stylelint/stylelint#3840]
> Clearly describe the bug

"shorthand-property-no-redundant-values" not throwing error for redundant negative values.

> Which rule, if any, is the bug related to?

`shorthand-property-no-redundant-values`

> What CSS is needed to reproduce the bug?

e.g.

```css
.criteriadesc{
  margin: 0 -20px 0 -20px;
  padding:10px 20px;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "shorthand-property-no-redundant-values": true
  }
}
```

> Which version of stylelint are you using?

e.g. `9.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No , it a standard CSS.

> What did you expect to happen?

Warnings should be flagged

> What actually happened (e.g. what warnings or errors did you get)?

No warnings were flagged.

**Comment by jeddy3** [comment]
@ssivanatarajan Thanks for the report and for using the template.

I can reproduce this bug. I'm surprised it's not come up before.

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by ssivanatarajan** [comment]
@jeddy3  , sure i will work on this.
